### DISCUSSION TITLE
Webcrape latest version

### DISCUSSION THREAD

**Discussion by MichaelKlinteberg** [git-for-windows/git#4246] [Q&A] [answered]
Why do you make it so difficult to webscrape latest version of Git for Windows?
Perhaps you can provide with Powerhell example how to find latest version, download and install/update Git 64 for Windows from URL
https://github.com/git-for-windows/git/releases/latest

**Comment by MichaelKlinteberg** [comment]
I found the API. 
Not the full solution. But it's a start

```
$uri = "https://api.github.com/repos/git-for-windows/git/releases/latest"
$webresponse = Invoke-WebRequest -UseBasicParsing $uri
$Json = $webresponse | ConvertFrom-Json
$download =  $Json.assets.browser_download_url -like '*Git-*-64-bit.exe'
```

**Reply by MichaelKlinteberg** [reply]
Above post removed the asterix
'asterixGit-asterix-64-bit.exe'

**Comment by dscho** [comment]
You would not happen to look for https://github.com/git-for-windows/git-for-windows.github.io/blob/main/latest-64-bit-installer.url, perchance? This is easily accessible via https://gitforwindows.org/latest-64-bit-installer.url

Also, you might find it more productive to ask in as kind a fashion as you would like to be answered.

**Reply by MichaelKlinteberg** [reply]
Dscho
Thank you for your answer and remind me that trolls like me should be more kind in asking questions.

The internet browser I use do not trust the download you provided. For now I rather webscrape the latest version from 
https://api.github.com/repos/git-for-windows/git/releases/latest
I can easy extract latest 64 bit from above url.

**Comment by MichaelKlinteberg** [comment, marked as answer]
```
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
$uri = "https://api.github.com/repos/git-for-windows/git/releases/latest"
$webresponse = Invoke-WebRequest -UseBasicParsing $uri
$Json = $webresponse | ConvertFrom-Json
$download =  $Json.assets.browser_download_url -like '*Git-*-64-bit.exe'
Start-BitsTransfer -Source $($download[0]) -Description $($json.name) -Destination ".\GitInstaller.exe"
Start-Process -FilePath ".\GitInstaller.exe" -ArgumentList "/SP- /VERYSILENT /SUPPRESSMSGBOXES /ALLUSERS /NORESTART /FORCECLOSEAPPLICATIONS /LOG=$env:TEMP\GitInstall.log" -Wait
& 'C:\Program Files\Git\cmd\git.exe' lfs install
```
